scripts/data_files/driver_templates/psa_crypto_driver_wrappers.c.jinja

```diff
@@ -19,43 +19,44 @@
  *  limitations under the License.
  */
 
+
+/* BEGIN-common headers */
 #include "common.h"
 #include "psa_crypto_aead.h"
 #include "psa_crypto_cipher.h"
 #include "psa_crypto_core.h"
 #include "psa_crypto_driver_wrappers.h"
 #include "psa_crypto_hash.h"
 #include "psa_crypto_mac.h"
-
 #include "mbedtls/platform.h"
+/* END-common headers */
 
 #if defined(MBEDTLS_PSA_CRYPTO_C)
 
+/* BEGIN-driver headers */
 #if defined(MBEDTLS_PSA_CRYPTO_DRIVERS)
-
-/* Include test driver definition when running tests */
-#if defined(PSA_CRYPTO_DRIVER_TEST)
-#ifndef PSA_CRYPTO_DRIVER_PRESENT
-#define PSA_CRYPTO_DRIVER_PRESENT
-#endif
-#ifndef PSA_CRYPTO_ACCELERATOR_DRIVER_PRESENT
-#define PSA_CRYPTO_ACCELERATOR_DRIVER_PRESENT
+{% for driver in drivers -%}
+/* Headers for {{driver.prefix}} {{driver.type}} driver */
+{% if driver.dependency is defined -%}
+#if {{ driver.dependency }}
+{% endif -%}
+{% for header in driver.headers -%}
+#include "{{ header }}"
+{% endfor %}
 #endif
```

Comment:
I don't think this adds much value as a macro since the depends_on property is different for capabilities and for header files.
This runs only once for headers, so having a macro for that is quite pointless, also the capabilities are triggered using independent caller templates, so macros there for this purpose also do not add much value.

Fixed the endif conditionals.

Reply:
TODO: Should possibly track improvements to catch these kind of errors with some dummy test drivers. Ideally anything with a base minimal set of default json params , this cannot be easily done with the current test drivers